### PR TITLE
Closes #2167 AZ Person 'Link to external bio' is not currently implemented

### LINKED ISSUE #2167: AZ Person "Link to external bio" is not currently implemented
## Problem/Motivation
While getting the president's site ready for launch we noticed that people in the grid view were not linking to their external bio pages.

We found that the az_card display mode does not use the "link to external bio" field.

On the president's site, I changed the az_card display to use `[node:az-canonical-url]`on `/admin/structure/types/manage/az_person/display/az_card` under `group_link`
image

The above action fixed the issue.

### To Reproduce

1. Go to `/node/add/az_person`
2. Fill out fields as you would normally.
3. Click on 'Summary Fields'
image

4. Add a link to a separate (external) profile.
5. Save
6. Go to `/people`
7. See link to external bio not being used.

## Proposed resolution
Use `[node:az-canonical-url]` as a custom URL for the link group on card or other listing view modes.

### PR DESCRIPTION
**Pull request by trackleft**
## Problem/Motivation
While getting the president's site ready for launch we noticed that people in the grid view were not linking to their external bio pages.

## Proposed resolution
Use `[node:az-canonical-url]` as a custom URL for the link group on card or other listing view modes.

## Description
We found that the az_card display mode does not use the "link to external bio" field.

On the president's site, I changed the az_card display to use `[node:az-canonical-url]`on `/admin/structure/types/manage/az_person/display/az_card` under `group_link`
image

The above action fixed the issue.

## Related issues
Closes #2167 
Blocked by https://github.com/az-digital/az_quickstart/pull/2077

## How to test

1. Go to `/node/add/az_person`
2. Fill out fields as you would normally.
3. Click on 'Summary Fields'
image

4. Add a link to a separate (external) profile.
5. Save
6. Go to `/people`
7. See link to external bio being used.

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
   - [ ] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
   - [ ] Adding experimental module
   - [ ] Update experimental module
- **Minor release changes**
   - [ ] New feature
   - [x] Breaking or visual change to existing behavior
   - [ ] Upgrade experimental module to stable
   - [ ] Enable existing module by default or database update
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- **Other or unknown**
   - [ ] Other or unknown

### Drupal core
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major or minor level update
- **Other or unknown**
   - [ ] Other or unknown

### Drupal contrib projects
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major level update
- **Other or unknown**
   - [ ] Other or unknown

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by bberndt-uaz — DISMISSED** [review]
Before we [converted AZ Person to use the Drupal UI](https://github.com/az-digital/az_quickstart/pull/1989), the Card display linked to the external bio page from both the card image and the card title. I recommend that we add another Link field group around the Smart Title element and uncheck the "Linked to entity" setting so that we can use `node:az-canonical-url` in the field group instead. Then the external bio link will be accessible even if the person node doesn't happen to have a photo selected.

I recommend making the same updates to the Row display as well.

**Comment by trackleft** [comment]
OK, that sounds good @bberndt-uaz , also, let's also update the [grid modal display ](https://github.com/az-digital/az_quickstart/pull/2077) to check if an external link is being used, and if so, use the overridden link.

**Comment by trackleft** [comment]
FYI, modal grid is a separate view mode, and does not support external link, and I was wondering if we should?

File: modules/custom/az_person/config/install/core.entity_view_display.node.az_person.az_card.yml [hunk 1]
```diff
@@ -103,10 +103,25 @@ third_party_settings:
...
+      region: content
+      weight: 12
+      format_type: link
+      format_settings:
+        classes: ''
+        show_empty_fields: false
+        id: ''
+        target: custom_uri
+        custom_uri: '[node:az-canonical-url]'
+        target_attribute: default
```
Comment (bberndt-uaz) [inline modules/custom/az_person/config/install/core.entity_view_display.node.az_person.az_card.yml:120]: My suggestion from my comment was not quite right, as it causes the heading style to override the link color:

image

To fix this and match the original HTML, I added a "Title Heading" field group around the link. Hopefully this suggestion works:

```suggestion
    group_title_heading:
      children:
        - group_title_link
      label: 'Title Heading'
      parent_name: group_card_body
      region: content
      weight: 1
      format_type: html_element
      format_settings:
        classes: 'h5 mt-0'
        show_empty_fields: false
        id: ''
        element: h3
        show_label: false
        label_element: h3
        label_element_classes: ''
        attributes: ''
        effect: none
        speed: fast
    group_title_link:
      children:
        - smart_title
      label: 'Title Link'
      parent_name: group_title_heading
      region: content
      weight: 1
      format_type: link
      format_settings:
        classes: ''
        show_empty_fields: false
        id: ''
        target: custom_uri
        custom_uri: '[node:az-canonical-url]'
        target_attribute: default
```

File: modules/custom/az_person/config/install/core.entity_view_display.node.az_person.az_card.yml [hunk 2]
```diff
@@ -103,10 +103,25 @@ third_party_settings:
...
   smart_title:
     enabled: true
     settings:
-      smart_title__link: true
+      smart_title__link: false
```
Comment (bberndt-uaz) [inline modules/custom/az_person/config/install/core.entity_view_display.node.az_person.az_card.yml:124]: To complete the change started with the commit suggestion above, the `smart_title__tag` should be set to `''` and `smart_title__classes` can be set to `{  }` in the lines below.

File: modules/custom/az_person/config/install/core.entity_view_display.node.az_person.az_card.yml [hunk 3]
```diff
@@ -103,10 +103,25 @@ third_party_settings:
...
       smart_title__tag: h3
       smart_title__classes:
         - h5
```
Comment (trackleft) [inline modules/custom/az_person/config/install/core.entity_view_display.node.az_person.az_card.yml:127]:
```suggestion
      smart_title__classes: {}
```

**Comment by bberndt-uaz** [comment]
The card view looks good! For the row view, it looks like the Smart Title "Linked to entity" box needs to be unchecked since both links are present:

```
<h3 class="h5 mt-0">
  <a href="http://quickstart.arizona.edu" class="field-group-link"></a>
  <a href="/person/brian-test-person" hreflang="en"><span class="field field--name-title field--type-string field--label-hidden">Brian Test Person</span>
  </a>
</h3>
```

**Comment by bberndt-uaz** [comment]
The Row view is linking to the external bio links correctly, but unfortunately the person node title needs the same wrapping HTML element as the grid view to fix the link color. Sorry!

image

**Review by bberndt-uaz — APPROVED** [review]

**Review by tadean — APPROVED** [review]

**Review by joeparsons — APPROVED** [review]

**Pull request merged**